pkg/storage/url/urldownload/storage.go

```diff
@@ -0,0 +1,123 @@
+package urldownload
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"path/filepath"
+
+	"github.com/filecoin-project/bacalhau/pkg/config"
+	"github.com/filecoin-project/bacalhau/pkg/storage"
+	"github.com/filecoin-project/bacalhau/pkg/system"
+	"github.com/go-resty/resty/v2"
+	"github.com/rs/zerolog/log"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// a storage driver runs the downloads content
+// from a public URL source and copies it to
+// a local directory in preparation for
+// a job to run - it will remove the folder/file once complete
+
+type StorageProvider struct {
+	LocalDir   string
+	HTTPClient *resty.Client
+}
+
+func NewStorageProvider(cm *system.CleanupManager) (*StorageProvider, error) {
+	// TODO: consolidate the various config inputs into one package otherwise they are scattered across the codebase
+	dir, err := ioutil.TempDir(config.GetStoragePath(), "bacalhau-url")
+	if err != nil {
+		return nil, err
+	}
+
+	client := resty.New()
+	// Setting output directory path, If directory not exists then resty creates one
+	client.SetOutputDirectory(dir)
+
+	storageHandler := &StorageProvider{
+		HTTPClient: client,
+		LocalDir:   dir,
+	}
+
+	log.Debug().Msgf("URL download driver created with output dir: %s", dir)
+	return storageHandler, nil
+}
+
+func (sp *StorageProvider) IsInstalled(ctx context.Context) (bool, error) {
+	_, span := newSpan(ctx, "IsInstalled")
+	defer span.End()
+	return true, nil
+}
+
+func (sp *StorageProvider) HasStorageLocally(ctx context.Context, volume storage.StorageSpec) (bool, error) {
+	_, span := newSpan(ctx, "HasStorageLocally")
+	defer span.End()
+	return false, nil
```

Comment:
I was about to comment "this should be true otherwise job selection policy won't pick up the job"

but then it struck me that the datra will ALWAYS be remote because it's a URL so good job :-)